packages/x-charts-pro/src/context/CartesianProviderPro/CartesianProviderPro.tsx

```diff
@@ -3,26 +3,55 @@ import {
   useDrawingArea,
   useSeries,
   CartesianContext,
-  CartesianContextProviderProps,
+  CartesianProviderProps,
   cartesianProviderUtils,
   useXExtremumGetter,
   useYExtremumGetter,
 } from '@mui/x-charts/internals';
 import { useZoom } from '../ZoomProvider/useZoom';
+import { createAxisFilterMapper } from './createAxisFilterMapper';
 
 const { computeValue } = cartesianProviderUtils;
 
-export interface CartesianContextProviderProProps extends CartesianContextProviderProps {}
+export interface CartesianProviderProProps extends CartesianProviderProps {}
 
-function CartesianContextProviderPro(props: CartesianContextProviderProProps) {
+function CartesianProviderPro(props: CartesianProviderProProps) {
   const { xAxis, yAxis, dataset, children } = props;
 
   const formattedSeries = useSeries();
   const drawingArea = useDrawingArea();
-  const { zoomData } = useZoom();
+  const { zoomData, options } = useZoom();
   const xExtremumGetters = useXExtremumGetter();
   const yExtremumGetters = useYExtremumGetter();
 
+  const zoomFilter = React.useMemo(() => {
+    const xMapper = createAxisFilterMapper({
+      zoomData,
+      extremumGetter: xExtremumGetters,
+      formattedSeries,
+    });
+
+    const yMapper = createAxisFilterMapper({
+      zoomData,
+      extremumGetter: yExtremumGetters,
+      formattedSeries,
+    });
+
+    const xFilters = xAxis.map(xMapper).filter((f) => f !== null);
+
+    const yFilters = yAxis.map(yMapper).filter((f) => f !== null);
+
+    if (xFilters.length === 0 && yFilters.length === 0) {
+      return undefined;
+    }
+
+    // Filters are applied on the reverse axis, so the naming below is correct.
+    return {
+      xFilters: (index: number) => yFilters.every((f) => f(index)),
+      yFilters: (index: number) => xFilters.every((f) => f(index)),
+    };
```

Comment:
I've I've changed it into a function instead in order to unify how we select the filters in a single place, this should make it clearer where we can improve the logic in the future, instead of duplicating the logic in every getter 😅

Reply:
Does not seems to work with continuouse scales

The `filter` function is getting values from the axis it computes extremums. But it should filter according to the axis with filterMode dsicad. Here are two examples (codesandbox does not manage to fetch pro package from the PR)

The bug is the following:

- the x-axis goes from 0 to 10 and has the filter discard
- the y-axis goes from 100 to 1,000 (just such that we can easily distinguish the values)

When you zoom, the filter will say for example that x-axis values need to be between 2 and 8 (that part is correct)

But the filter function is getting values from the y-axis, So none of them pass the filter.

When computing the extremum for the y-axis it's the x-axis value that should be provided to the filter. 


## Potential fixes

### For line charts

For charts with an axis with `data` property, you can simply ignore the `value` and rely exclusively on the `dataIndex`.

### For scatter chart

To keep your current solution, you could replace the value by an object `{x, y}` such that a filter set on the x-axis uses the `x` and a filter set on the y-axis uses the `y`.

```jsx
import * as React from 'react';
import { LineChartPro } from '@mui/x-charts-pro';
import { ScatterChartPro } from '@mui/x-charts-pro';

export default function AreaBaseline() {
  return (
    <>
      <LineChartPro
        xAxis={[{ data: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], zoom: { filterMode: 'discard' } }]}
        series={[
          {
            data: [10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
            area: true,
            baseline: 'min',
          },
        ]}
        width={500}
        height={300}
      />
      <ScatterChartPro
        xAxis={[{ data: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], zoom: { filterMode: 'discard' } }]}
        series={[
          {
            data: [
              { id: 0, x: 1, y: 100 },
              { id: 1, x: 1, y: 500 },
              { id: 2, x: 5, y: 200 },
              { id: 3, x: 10, y: 300 },
              { id: 4, x: 15, y: 400 },
              { id: 5, x: 20, y: 500 },
            ],
          },
        ]}
        width={500}
        height={300}
      />
    </>
  );
}
```